tests/Internal/Codebase/ClassLikesTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+declare(strict_types=1);
+
+namespace Psalm\Tests\Internal\Codebase;
+
+use Psalm\Codebase;
+use Psalm\Internal\Codebase\ClassLikes;
+use Psalm\Internal\Codebase\Reflection;
+use Psalm\Internal\Codebase\Scanner;
+use Psalm\Internal\Provider\ClassLikeStorageProvider;
+use Psalm\Internal\Provider\FileReferenceProvider;
+use Psalm\Internal\Provider\FileStorageProvider;
+use Psalm\Internal\Provider\Providers;
+use Psalm\Internal\Provider\StatementsProvider;
+use Psalm\Progress\VoidProgress;
+use Psalm\Storage\ClassLikeStorage;
+use Psalm\Tests\TestCase;
+
+final class ClassLikesTest extends TestCase
+{
+    /**
+     * @var ClassLikes
+     */
+    private $classLikes;
+
+    /**
+     * @var ClassLikeStorageProvider
+     */
+    private $storageProvider;
+
+    /**
+     * @var FileReferenceProvider
+     */
+    private $fileReferenceProvider;
+
+    /**
+     * @var StatementsProvider
+     */
+    private $statementsProvider;
+
+    /**
+     * @var Scanner
+     */
+    private $codebaseScanner;
+
+    /**
+     * @var Codebase
+     */
+    private $codebase;
+
+    /**
+     * @var Providers
+     */
+    private $providers;
+
+    /**
+     * @var VoidProgress
+     */
+    private $progress;
+
+    /**
+     * @var FileStorageProvider
+     */
+    private $fileStorageProvider;
+
+    /**
+     * @var Reflection
+     */
+    private $reflection;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $this->storageProvider = new ClassLikeStorageProvider();
+        $this->fileReferenceProvider = new FileReferenceProvider();
+        $this->statementsProvider = new StatementsProvider($this->file_provider);
+        $this->providers = new Providers($this->file_provider);
```

Comment:
After `parent::setUp()` call you already have a fully initialized `ProjectAnalyzer` that is reinitialized for every test method. You can use that to obtain `ClassLikes` and `ClassLikeStorageProvider`:

```suggestion
        $this->classlikes = $this->project_analyzer->getCodebase()->classlikes;
        $this->storage_provider = $this->project_analyzer->getCodebase()->classlike_storage_provider;
```

Or just use those directly in the test method.

Reply:
Oh nice! Wasn't aware of that. Thanks!